common/timer/local_gate.go

```diff
@@ -0,0 +1,123 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package timer
+
+import (
+	"time"
+
+	"go.temporal.io/server/common/clock"
+)
+
+type (
+	LocalGate interface {
+		Gate
+	}
+
+	LocalGateImpl struct {
+		// the channel which will be used to proxy the fired timer
+		fireChan  chan struct{}
+		closeChan chan struct{}
+
+		timeSource clock.TimeSource
+
+		// the actual timer which will fire
+		timer *time.Timer
+		// variable indicating when the above timer will fire
+		nextWakeupTime time.Time
+	}
+)
+
+// NewLocalGate create a new timer gate instance
+func NewLocalGate(timeSource clock.TimeSource) LocalGate {
```

Comment:
github is not able to figure out this is old code moved around? I hate that we lost the history of the code.

Reply:
not sure ...

Reply:
maybe due to TimerGate => timer.Gate